types/signed_header.go

```diff
@@ -7,51 +7,62 @@ import (
 
 	"github.com/celestiaorg/go-header"
 	"github.com/cometbft/cometbft/crypto/ed25519"
+	cmtypes "github.com/cometbft/cometbft/types"
 )
 
-func (sH *SignedHeader) New() header.Header {
+// SignedHeader combines Header and its Commit.
+//
+// Used mostly for gossiping.
+type SignedHeader struct {
+	Header
+	Commit     Commit
+	Validators *cmtypes.ValidatorSet
+}
+
+func (sh *SignedHeader) New() *SignedHeader {
 	return new(SignedHeader)
 }
 
-func (sH *SignedHeader) IsZero() bool {
-	return sH == nil
+func (sh *SignedHeader) IsZero() bool {
+	return sh == nil
 }
 
 var (
-	ErrAggregatorSetHashMismatch        = errors.New("aggregator set hash in signed header and hash of validator set do not match")
-	ErrSignatureVerificationFailed      = errors.New("signature verification failed")
-	ErrNoProposerAddress                = errors.New("no proposer address")
-	ErrLastHeaderHashMismatch           = errors.New("last header hash mismatch")
-	ErrLastCommitHashMismatch           = errors.New("last commit hash mismatch")
-	ErrNewHeaderTimeBeforeOldHeaderTime = errors.New("new header has time before old header time")
-	ErrNewHeaderTimeFromFuture          = errors.New("new header has time from future")
+	ErrNonAdjacentHeaders     = errors.New("non-adjacent headers")
+	ErrNoProposerAddress      = errors.New("no proposer address")
+	ErrLastHeaderHashMismatch = errors.New("last header hash mismatch")
+	ErrLastCommitHashMismatch = errors.New("last commit hash mismatch")
 )
 
-func (sH *SignedHeader) Verify(untrst header.Header) error {
-	// Explicit type checks are required due to embedded Header which also does the explicit type check
-	untrstH, ok := untrst.(*SignedHeader)
-	if !ok {
-		// if the header type is wrong, something very bad is going on
-		// and is a programmer bug
-		panic(fmt.Errorf("%T is not of type %T", untrst, untrstH))
-	}
+func (sh *SignedHeader) Verify(untrstH *SignedHeader) error {
+	// TODO(@Wondertan):
+	//  We keep this check because of how unit tests are structured where TestVerify tests both ValidateBasic and Verify.
+	//  While the check is redundant as go-header ensures untrustH passed ValidateBasic before.
+	//  Decoupling TestVerify into TestValidateBasic and TestVerify would allow to remove this check.
```

Comment:
This may be worth to be an issue, defer to you

Reply:
Made an issue here: https://github.com/rollkit/rollkit/issues/1189